vault/src/execution.rs

```diff
@@ -237,16 +239,135 @@ impl Request {
+                    match btc_rpc.fee_rate(txid) {
+                        Ok(current_fee) => {
+                            if x >= current_fee {
+                                Ok(Some((current_fee, x)))
+                            } else {
+                                Ok(None)
+                            }
+                        }
+                        Err(e) => {
+                            tracing::debug!("Failed to get fee_rate: {}", e);
+                            Ok(None)
+                        }
+                    }
+                })
+                .filter_map(|x| async {
+                    match btc_rpc.is_in_mempool(txid_copy) {
+                        Ok(false) => {
+                            //   if not in mempool anymore, don't propagate the event (even if it is an error)
+                            tracing::debug!("Txid not in mempool anymore...");
+                            None
+                        }
+                        Ok(true) => {
+                            tracing::debug!("Txid is still in mempool...");
+                            Some(x)
+                        }
+                        Err(e) => Some(Err(e.into())),
+                    }
+                });
+
+            let wait_for_transaction_metadata = btc_rpc.wait_for_transaction_metadata(txid, num_confirmations);
+            futures::pin_mut!(subscription);
+
+            let mut metadata_fut = wait_for_transaction_metadata;
+
+            // The code below looks a little bit complicated but the idea is simple:
+            // we keep waiting for inclusion until it's either included in the bitcoin chain,
+            // or we successfully bump fees
+            let tx_metadata = loop {
+                match futures::future::select(metadata_fut, subscription.next()).await {
+                    Either::Left((result, _)) => break result?,
+                    Either::Right((None, _)) => return Err(Error::RuntimeError(RuntimeError::ChannelClosed)),
+                    Either::Right((Some(Err(x)), continuation)) => {
+                        tracing::warn!("Received an error from the fee rate subscription: {x}");
+                        // continue with the unchanged fee rate
+                        metadata_fut = continuation;
+                    }
+                    Either::Right((Some(Ok((old_fee, new_fee))), continuation)) => {
+                        tracing::debug!("Attempting to bump fee rate from {old_fee} to {new_fee}...");
+                        match self.bump_fee(btc_rpc, txid, new_fee).await {
+                            Ok(new_txid) => {
+                                tracing::info!("Bumped fee rate. Old txid = {txid}, new txid = {new_txid}");
+                                txid = new_txid;
+                                continue 'outer;
+                            }
+                            Err(Error::BitcoinError(x)) if x.rejected_by_network_rules() => {
```

Comment:
Will this catch the case that the Vault can not afford the fee increase?

Reply:
Hmm that's a good one, probably not. We should add an explicit catch for that I guess

Reply:
I decided to not abort at all when getting errors. We'll just wait for the original tx